edgelet/edgelet-http-workload/src/server/cert/mod.rs

```diff
@@ -167,15 +167,26 @@ fn generate_key_and_csr(
     let mut extensions = openssl::stack::Stack::new()?;
     extensions.push(extended_key_usage)?;
 
-    if props.san_entries().is_some() {
+    if props.dns_san_entries().is_some() || props.ip_entries().is_some() {
         let mut subject_alt_name = openssl::x509::extension::SubjectAlternativeName::new();
-        props
-            .san_entries()
-            .expect("san entries unexpectedly empty")
-            .iter()
-            .for_each(|s| {
-                subject_alt_name.dns(s);
-            });
+        if props.dns_san_entries().is_some() {
```

Comment:
```diff
-if some_option.is_some() { some_option.expect("...").iter().for_each(...); }`
+some_option.iter().flatten().for_each("...");
```

Reply:
`flatten()` doesn't work because the option is `Option<&[String]>` and iter is `Iter<&[String]>`. Flatten converts it to a `&&[String]` which isn't an `Iterator`. Any other suggestions to extract the `[String]` from an `Option<&[String]>` inline?

Reply:
If it's an `Option<&T>` then use `.into_iter()` instead of `.iter()`